ballista-cli/Cargo.toml

```diff
@@ -29,12 +29,12 @@ rust-version = "1.59"
 readme = "README.md"
 
 [dependencies]
-ballista = { path = "../ballista/client", version = "0.9.0", features = [
-    "standalone",
-] }
+ballista = { path = "../ballista/client", version = "0.9.0", features = ["standalone"] }
+ballista-core = { path = "../ballista/core", version = "0.9.0" }
 clap = { version = "3", features = ["derive", "cargo"] }
-datafusion = "14.0.0"
-datafusion-cli = "14.0.0"
+dashmap = "5.4.0"
+datafusion = { git = "https://github.com/apache/arrow-datafusion.git", rev = "7b5842b91ebd00a2c7f894fcad797bea68a56d0f" }
```

Comment:
If we need features added since 14.0.0, then we either need to delay merging this PR until after we release Ballista 0.10.0, or we can merge now and then wait for DataFusion 15.0.0 before releasing Ballista 0.10.0. I don't have a strong preference either way.

Reply:
I see we have done the git-ref thing in the past, so there's precedent.

For FlightSql in particular, I've had to wait a long time for arrow to release, then datafusion, then ballista. I've been rebasing changes for months and our fork continues to have changes that are still waiting (in this case on delta-rs, but now they are way behind datafusion).

For these reasons, I'd strongly prefer to do a ballista release shortly after the datafusion release, but then go back to git-ref tracking, so we can keep PRs flowing. 

Just my $0.02